code/_core/obj/projectile/_projectile.dm

```diff
@@ -203,6 +204,16 @@
 		TE.maptext = "[steps_current]"
 		TE.alpha = 200
 
+	for(var/obj/projectile/other_P in new_loc.contents) //Handles P on P collisions
+		if(projectile_should_collide(other_P,old_loc,new_loc) || other_P.projectile_should_collide(src,old_loc,new_loc))
+			if(other_P.on_projectile_hit(src,old_loc,new_loc)) //Checks if the other projectile should do something to this one.
+				other_P.penetrations_left--
+				//Dont return anything here, it may have more to its life, but note it did hit something.
+			if(on_projectile_hit(other_P,old_loc,new_loc)) //Checks if this projectile should do something to the other one.
+				penetrations_left--
+				if(penetrations_left < 0)
+					return FALSE //This projectile has served its purpose.
+
```

Comment:
This code unfortunately will not work as intended because of how projectiles work.

When a projectile is created, it does not actually move. It stays in one location and the movement that it has is completely simulated to bypass byond's movement to reduce the amount of lag projectiles have.

The only feasible way to add this feature is to make it so that the pyro airblast creates a special type of /obj/ that projectiles can collide with via the projectile_should_collide() proc so that moving projectiles can detect this /obj/.

Reply:
Believe it or not, not only does this code work, it works as I intended it to!
Because of it being coded in an either-or sense, it doesnt care if a projectile enters its tile or if it enters a tile with a projectile, the airblast detects it and reflects properly. This also allows anyone to make their projectile, say, absorb other projectiles to make itself stronger. I've tested this for a few rounds, and it works superbly; I've noticed no major performance degradation from this change.

Though if you're not satisfied with this solution, let me know and I will gladly find another solution.

Reply:
This might create issues where you can airblast bullets where they spawned and those bullets will be reflected even though it shouldn't be. I highly recommend doing my solution.

Reply:
Took me a moment but I get you now, will change solution